include/frequency_cam/frequency_cam.h

```diff
@@ -39,29 +40,77 @@ class FrequencyCam : public event_array_codecs::EventProcessor
   // ------------- inherited from EventProcessor
   inline void eventCD(uint64_t sensor_time, uint16_t ex, uint16_t ey, uint8_t polarity) override
   {
+    // If the first time stamp is > 15s, there is an offset which we subtract every time.
+    if (!initialize_time_stamps_) {
+      initialize_time_stamps_ = true;
+      if (sensor_time > 15000000000) {
+        fix_time_stamps_ = true;
+      }
+    }
+    if (fix_time_stamps_) {
+      // Offset taken from:
+      // https://docs.prophesee.ai/stable/data/encoding_formats/evt3.html#evt-time-high
+      sensor_time -= 16777215000;
+    }
     Event e(shorten_time(sensor_time), ex, ey, polarity);
     updateState(&state_[e.y * width_ + e.x], e);
     lastEventTime_ = e.t;
+    lastEventTimeNs_ = sensor_time;
     eventCount_++;
+    eventTimesNs_.emplace_back(sensor_time);
+  }
+  void eventExtTrigger(uint64_t sensor_time, uint8_t edge, uint8_t /*id*/) override
```

Comment:
It does a bit more. We also check, if the external trigger events switch their polarity to ensure we don't miss an external trigger event. More importantly, we set `sensor_time_` to the timestamps of the external trigger event and `hasValidTime_` to `true` so that we can use them in `frequency_cam.cpp:177` to check for the CD event with the time stamps closest to the one of the external trigger event.
Please let me know, if my explanation lacks some clarity.